File: backend/app/schemas/drawing.py
# app/schemas/drawing.py
from datetime import date
from typing import Optional, List
from pydantic import BaseModel


# ---- DrawingCollection schema ----
class DrawingCollection(BaseModel):
    drawing_id: int
    collection_id: int
    position: Optional[int] = None 


# ---- Collection schema ----
class Collection(BaseModel):
    id: int
    name: str
    description: Optional[str] = None
    type: Optional[str] = None 


# ---- Drawing schema ----
class Drawing(BaseModel):
    id: int
    title: str
    description: Optional[str] = None
    file_url: str
    thumbnail_url: Optional[str] = None
    drawing_date: Optional[date] = None
    is_favorite: bool = False
    is_public: bool = False
    collections: Optional[List[Collection]] = [] 

File: src/types/drawing.ts

class Drawing {
    "id": number
    "title": string | null
    "description": string | null
    "file_url": string
    "thumbnail_url": string | null
    "drawing_date": string | null
    "is_favorite": boolean
    "collections": number[]
}

class Collection {
    "id": number
    "name": string
    "description": string
    "type": "videogame" | "sketchbook" | "franchise" | "challenge" | "year" | "project" | "character" | "other"
    "drawings": number[]
}
export type { Drawing, Collection}